### PR TITLE
Backend communications

### PR DESCRIPTION
**Pull request by CFIALeronB**
- Creating dedicated api.js file that contains api call functions.
- Renaming function fecthData to PingBackend.
- Moved PingBackend to api.js
- Implemented on screen alert if empty query response is invalid or if backend environment variable URL is empty or NULL.
- Moved and renamed api unit test file.

### CODE REVIEW

File: src/pages/home/Home.js [hunk 1]
```diff
@@ -2,27 +2,38 @@ import { SearchBar } from '../../components/SearchBar';
...
+
+    useEffect(() => {
+        // Check if the REACT_APP_BACKEND_URL is set
+        if (process.env.REACT_APP_BACKEND_URL) {
+            PingBackend().catch((error) => {
+                console.error("Error: ", error);
+                setAlertMessage("Request failed"); // Set the alert message on error
+            });
+        } else {
+            console.error('The REACT_APP_BACKEND_URL environment variable is not set or empty.');
```
Comment (rngadam) [inline src/pages/home/Home.js:19]: also an alert for the user "Backend URL is not set, frontend is misconfigured"

File: src/api/api.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        if(response.ok) {
+            console.log("Active Server Connection");
+        } else{
+            throw new Error("Request failed");
+        }
+        
+    }catch (error){
+        console.error("Error: ", error);
+    }
+}
```
Comment (rngadam) [inline src/api/api.js:23]: the little red marker at the end of files mean you are missing a new line

File: src/api/api.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+                'Content-Type':'application/json'
+            },
+            body: JSON.stringify({
+                query: ""
+            }),
+        });
+
+        if(response.ok) {
+            console.log("Active Server Connection");
+        } else{
```
Comment (rngadam) [inline src/api/api.js:16]: your curly braces have inconsistent spacing. do:

```
} else {
```

**Review by rngadam — CHANGES_REQUESTED** [review]
see comments I left

File: src/api/api.js (hunk 3 above)
Comment (CFIALeronB) [reply src/api/api.js]: @rngadam Fixed code indention

File: src/api/api.test.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// Create a test case for the PingBackend function
+test('Testing Response Time', async () => {
+    // Mock the response from the API
+    // Set the expected response time
+    const expectedResponseTime = 10000; // Specify the desired response time in milliseconds
+    const url = process.env.REACT_APP_BACKEND_URL;
+
+    expect(url).not.toEqual("");
+
+    if(url != ""){
```
Comment (rngadam) [inline src/api/api.test.js:13]: spacing

File: src/api/useApiUtil.js [hunk 5]
```diff
@@ -23,23 +24,42 @@ const useApiUtil = (term) => {
                     console.log(resposeData);
```
Comment (rngadam) [inline src/api/useApiUtil.js:24]: misspelling

File: src/api/useApiUtil.js [hunk 6]
```diff
@@ -23,23 +24,42 @@ const useApiUtil = (term) => {
...
     return { data };
-
 }
 
-export default useApiUtil;
\ No newline at end of file
+export const PingBackend = async () => {
+    try{
+        const response = await fetch(process.env.REACT_APP_BACKEND_URL + "/search", {
+            method: "POST",
```
Comment (rngadam) [inline src/api/useApiUtil.js:47]: PingBackend(get_endpoint) 
    fetch(get_endpoint('search'));

**Review by rngadam — CHANGES_REQUESTED** [review]
fix minor formatting issues

**Review by rngadam — APPROVED** [review]
LGTM

**Pull request merged**